src/julia-syntax.scm

```diff
@@ -2396,7 +2367,7 @@
                      ,.(apply append rows)))
             `(call (top typed_vcat) ,t ,@a)))))
 
-   '|'|  (lambda (e) (expand-forms `(call adjoint ,(cadr e))))
+   '|'|  (lambda (e) (expand-forms `(call (core postfixapostrophize) ,(cadr e))))
```

Comment:
Use `top` not `core`, and put separate definitions inside each topmodule (Inference and Base) instead of in boot.jl. That'll fix the method overwrite issue.

Reply:
Thanks @vtjnash! Seems Jeff is addressing the warning in #25212 :).